`-no-entry-point` inc conjunction with `-no-crt` and `-build-mode:dll` produces linker error for dllmain

## Context

Please provide any relevant information about your setup. This is important in case the issue is not reproducible except for under certain conditions.
        Odin:    dev-2024-12-nightly:cf53404
        OS:      Windows 10 Enterprise N (version: 20H2), build 19042.1466   
        CPU:     Intel(R) Core(TM) i5-8400 CPU @ 2.80GHz
        RAM:     65470 MiB
        Backend: LLVM 18.1.8

## Expected Behavior

Compiling a empty dll without main and crt should just work.

## Current Behavior

Compiling an empty file with `-file -no-crt -no-entry-point -build-mode:dll` produces  
```
LINK : error LNK2001: unresolved external symbol _DllMainCRTStartup
```
and fails to produce the dll.

## Failure Information (for bugs)

### Steps to Reproduce

1. create an empty file`test.odin`
2. compile it with `odin build test.odin -file -no-crt -no-entry-point -build-mode:dll`
3. liker error

## Workaround

For Microsoft cl you can add `-extra-linker-flags:/NOENTRY` (which i expected to already be set according to the odin flags, but i appears not to be, which solved the issue.

--- a/src/linker.cpp
+++ b/src/linker.cpp
@@ -277,6 +277,9 @@ try_cross_linking:;
 
 			if (build_context.build_mode == BuildMode_DynamicLibrary) {
 				link_settings = gb_string_append_fmt(link_settings, " /DLL");
+				if (build_context.no_entry_point) {
+					link_settings = gb_string_append_fmt(link_settings, " /NOENTRY");
+				}
 			} else {
 				link_settings = gb_string_append_fmt(link_settings, " /ENTRY:mainCRTStartup");
 			}

